tools/ZarinPalGateway.php

```diff
@@ -0,0 +1,311 @@
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, CURLOPT_HTTPHEADER, array(
+            'Content-Type: application/json',
+            'Content-Length: ' . strlen($jsonData)
+        ));
+
+        $result = curl_exec($ch);
+        $err = curl_error($ch);
+        $result = json_decode($result, true);
+        curl_close($ch);
+
+        //if there is an cURL error
+        if ($err) {
+            $this->error = "cURL Error : " . $err;
+
+            return false;
+        } else {
+            //Check for gate result
+            if ($result["Status"] == 100) {
+                /*
+                 * If payment failed update new authority or creat new one
+                 */
+                $cart_id = Context::getContext()->cart->id;
+                if (PgzarinpalModel::cartIdExists($cart_id)) {
+                    $id_pgzarinpal = PgzarinpalModel::getIdByCartId($cart_id);
+                    $pgzarinpal = new PgzarinpalModel($id_pgzarinpal);
+
+                    $pgzarinpal->authority = $result["Authority"];
+                    $authority_result = $pgzarinpal->update();
+                } else {
+                    $pgzarinpal = new PgzarinpalModel();
+
+                    $pgzarinpal->cart_id = (int)$cart_id;
+                    $pgzarinpal->authority = $result["Authority"];
+                    $authority_result = $pgzarinpal->save();
+                }
+
+                /*
+                 * Redirect to payment page if authority saved
+                 */
+                if ($authority_result) {
+                    //redirect to gate
+                    header("Location: https://{$this->sandbox}.zarinpal.com/pg/StartPay/{$result["Authority"]}{$this->zaringate}");
+                    return true;
+                } else {
+                    $this->error = $this->l("Failed to save payment info");
+
+                    return false;
+                }
+            } else {
+                $this->error = $result["Status"];
+
+                return false;
+            }
+        }
+    }
+
+    /**
+     * verify the payment request
+     * Return an error message on failure that you must get the error via getPaymentError()
+     * Return reference ID on success
+     *
+     * @param int $amount
+     * @param string $authority
+     *
+     * @return bool|string
+     */
+    public function verifyPayment($amount, $authority)
+    {
+        //Check that currency is IRR or IRT
+        if ($this->currency === false) {
+            $this->error = $this->l("Shop currency must be IRR or IRT");
+
+            return false;
+        }
+
+        //Convert IRR to IRT because ZarinPal excepts just IRT
+        if ($this->currency == "IRR") {
+            $amount = $this->convertToIrt($amount);
+        }
+
+        //Gate payment verification data
+        $data = [
+            'MerchantID' => $this->merchant_id,
+            'Authority' => $authority,
+            'Amount' => (int)$amount
```

Comment:
```suggestion
            'Amount' => (int) $amount
```

Reply:
We dont need space between (int) and variable